scripts/viz/tsne_viz.R

```diff
@@ -1,25 +1,47 @@
 # Pan-Cancer Variational Autoencoder
 # Gregory Way 2017
 #
-# Visualize t-SNE plots of VAE and raw RNAseq data features
+# Visualize t-SNE plots of VAE, ADAGE, and raw RNAseq data features
 
 library(dplyr)
 library(ggplot2)
 
+plot_tsne <- function(tsne_df) {
+  # Function to output tsne visualizations colored by cancer type for the given
+  # input data (compressed by either Tybalt or ADAGE or raw RNAseq data)
+  p <- ggplot(tsne_df, aes(x = `1`, y = `2`, color = acronym)) +
+    geom_point(size = 0.001) +
+    scale_colour_manual(limits = tcga_colors$`Study Abbreviation`,
+                        values = tcga_colors$`Hex Colors`,
+                        na.value = "black",
+                        breaks = palette_order) +
+    theme_classic() +
+    theme(legend.title = element_text(size = 8),
+          legend.text = element_text(size = 5),
+          legend.key.size = unit(0.5, "line")) +
+    guides(colour = guide_legend(override.aes = list(size = 0.1),
+                                 title = 'Cancer-Type'))
+  return(p)
+}
+
 # Load encodings file with matched clinical data to subset
 clinical_file <- file.path("data", "tybalt_features_with_clinical.tsv")
 clinical_df <- readr::read_tsv(clinical_file)
```

Comment:
Does the clinical_df need to include the gene expression data in addition to the sample id, or can this be the sample id's and cancer type and then you can apply "inner join" on the sample id?

Reply:
yeah, that is definitely an option - i had previously saved this file for use in a different script that required it, so I figured I would use it here!